torch/csrc/lazy/core/ir.h

```diff
@@ -65,7 +65,8 @@ using OutputMap = std::unordered_map<Output, T, Output::Hasher>;
 // Represents an input/operand for a Node object.
 struct TORCH_API Value {
   Value() = default;
-  /* implicit */ Value(NodePtr node, size_t index = 0) : node(std::move(node)), index(index) {}
+  /* implicit */ Value(NodePtr&& node, size_t index = 0) : node(std::move(node)), index(index) {}
+  /* implicit */ Value(const NodePtr& node, size_t index = 0) : node(node), index(index) {}
```

Comment:
could you explain what the difference ends up being when using the `const NodePtr` ctor with std::move at the call site, vs using the NodePtr&& ctor with std::move in the initializer list? 

Specifically since NodePtr is a shared_ptr type, i'd be curious to know if one method is doing more or less refcount bumping or allocations of new ptr classes.

Reply:
The first ctor is for the case where the caller don't bother to own the NodePtr. And the second one is for the case where the caller still want to owns the node, and thus the ownership get shared by copying the node in the init list here.

For the first one, ideally the refcount will stay the same while the second one will increment the refcount.

Reply:
Are you changing it so that the codegen nativefunctions use the second ctor, and make a copy+refcount bump, when it previously did a move? That sounds like the opposite of what we want.  But I'm probably misunderstanding.

Reply:
I'm changing it such that the code-gen uses the first one. But there are some existing code that relies on the second one's behavior (copy). That's why I didn't delete the second one.

Reply:
ohh i see now.  i didn't realize you changed NodePtr -> NodePtr&& for the first one.  Missed that.  Thanks!